Backport variable name used in dlsym typo correction.
Revert "Link libhts.so into plugins on Unix&macOS" on macOS.

diff --git a/plugin.c b/plugin.c
index dbbe03e..d5c1981 100644
--- a/plugin.c
+++ b/plugin.c
@@ -153,7 +153,7 @@ plugin_void_func *load_plugin(void **pluginp, const char *filename, const char *
         const char *basename = slash? slash+1 : filename;
         kputsn(basename, strcspn(basename, ".-+"), &symbolg);
 
-        *(void **) &sym = dlsym(lib, symbol);
+        *(void **) &sym = dlsym(lib, symbolg.s);
         free(symbolg.s);
         if (sym == NULL) goto error;
     }
diff --git a/Makefile b/Makefile
index 245b7a1..3bb21ad 100644
--- a/Makefile
+++ b/Makefile
@@ -310,8 +310,8 @@ hts-object-files: $(LIBHTS_OBJS)
 %.so: %.pico libhts.so
 	$(CC) -shared -Wl,-E $(LDFLAGS) -o $@ $< libhts.so $(LIBS) -lpthread
 
-%.bundle: %.o libhts.dylib
-	$(CC) -bundle -Wl,-undefined,dynamic_lookup $(LDFLAGS) -o $@ $< libhts.dylib $(LIBS)
+%.bundle: %.o
+	$(CC) -bundle -Wl,-undefined,dynamic_lookup $(LDFLAGS) -o $@ $< $(LIBS)
 
 %.cygdll: %.o libhts.dll.a
 	$(CC) -shared $(LDFLAGS) -o $@ $< libhts.dll.a $(LIBS)
